File: app/src/main/java/com/letianpai/robot/desktop/ui/view/AppSettingsNoticeView.java
package com.letianpai.robot.desktop.ui.view;

import android.content.Context;
import android.text.TextUtils;
import android.util.AttributeSet;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.annotation.Nullable;

import com.letianpai.robot.desktop.R;
import com.letianpai.robot.desktop.ui.activity.AppDefaultSettingActivity;
import com.letianpai.robot.desktop.ui.activity.AppModeSwitchActivity;

public class AppSettingsNoticeView extends RelativeLayout {
    private Context mContext;
    private BackButton backButton;
    private TextView enDes;
    private TextView zhDes;

    public AppSettingsNoticeView(Context context) {
        super(context);
        inits(context);
    }

    private void inits(Context context) {
        this.mContext = context;
        inflate(mContext, R.layout.app_setting_notice_view,this);
        initViews();
        setData();
        addListeners();

    }

    private void setData() {
        if (((AppDefaultSettingActivity) mContext).getAppMenuInfo() != null && !TextUtils.isEmpty(((AppDefaultSettingActivity) mContext).getAppMenuInfo().getEnDes())
                &&  !TextUtils.isEmpty(((AppDefaultSettingActivity) mContext).getAppMenuInfo().getEnDes())) {
            enDes.setText( ((AppDefaultSettingActivity) mContext).getAppMenuInfo().getEnDes());
            zhDes.setText( ((AppDefaultSettingActivity) mContext).getAppMenuInfo().getZhDes());

        } else {
            enDes.setText( mContext.getText(R.string.default_en));
            zhDes.setText( mContext.getText(R.string.default_zh));
        }
    }

    private void addListeners() {

        backButton.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                responseBack();
            }
        });

    }

    private void setCurrentRobotMode() {

    }

    private void openSettings() {

    }

    private void responseBack() {
        ((AppDefaultSettingActivity)mContext).finish();
    }

    private void initViews() {
        backButton = findViewById(R.id.back);
        enDes = findViewById(R.id.en_des);
        zhDes = findViewById(R.id.zh_des);
    }

    public AppSettingsNoticeView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        inits(context);
    }

    public AppSettingsNoticeView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        inits(context);
    }

    public AppSettingsNoticeView(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        inits(context);
    }
}

File: app/src/main/java/com/letianpai/robot/desktop/ui/view/AppSettingsNoticeView.kt
package com.letianpai.robot.desktop.ui.view

import android.content.Context
import android.text.TextUtils
import android.util.AttributeSet
import android.widget.RelativeLayout
import android.widget.TextView
import com.letianpai.robot.desktop.R
import com.letianpai.robot.desktop.ui.activity.AppDefaultSettingActivity

class AppSettingsNoticeView : RelativeLayout {
    private var mContext: Context? = null
    private var backButton: BackButton? = null
    private var enDes: TextView? = null
    private var zhDes: TextView? = null

    constructor(context: Context) : super(context) {
        inits(context)
    }

    private fun inits(context: Context) {
        this.mContext = context
        inflate(mContext, R.layout.app_setting_notice_view, this)
        initViews()
        setData()
        addListeners()
    }

    private fun setData() {
        if ((mContext as AppDefaultSettingActivity).appMenuInfo != null && !TextUtils.isEmpty((mContext as AppDefaultSettingActivity).appMenuInfo!!.enDes)
            && !TextUtils.isEmpty((mContext as AppDefaultSettingActivity).appMenuInfo!!.enDes)
        ) {
            enDes!!.text = (mContext as AppDefaultSettingActivity).appMenuInfo!!.enDes
            zhDes!!.text = (mContext as AppDefaultSettingActivity).appMenuInfo!!.zhDes
        } else {
            enDes!!.text = mContext!!.getText(R.string.default_en)
            zhDes!!.text = mContext!!.getText(R.string.default_zh)
        }
    }

    private fun addListeners() {
        backButton!!.setOnClickListener { responseBack() }
    }

    private fun setCurrentRobotMode() {
    }

    private fun openSettings() {
    }

    private fun responseBack() {
        (mContext as AppDefaultSettingActivity).finish()
    }

    private fun initViews() {
        backButton = findViewById(R.id.back)
        enDes = findViewById(R.id.en_des)
        zhDes = findViewById(R.id.zh_des)
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        inits(context)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    ) {
        inits(context)
    }

    constructor(
        context: Context,
        attrs: AttributeSet?,
        defStyleAttr: Int,
        defStyleRes: Int
    ) : super(context, attrs, defStyleAttr, defStyleRes) {
        inits(context)
    }
}
